### PR TITLE
Metrics context/thread fixed and javadoc

### PR DESCRIPTION
**Pull request by vietj**
@purplefox here it is.

### CODE REVIEW

File: src/main/java/io/vertx/core/http/impl/ServerConnection.java [hunk 1]
```diff
@@ -102,14 +102,17 @@
...
-    this.metric = metrics.connected(remoteAddress());
   }
 
   @Override
-  protected Object metric() {
+  protected synchronized Object metric() {
     return metric;
   }
 
+  void setMetric(Object metric) {
```
Comment (purplefox) [inline src/main/java/io/vertx/core/http/impl/ServerConnection.java]: not synchronized?

File: src/main/java/io/vertx/core/http/impl/ServerConnection.java [hunk 2]
```diff
@@ -102,14 +102,17 @@
     this.server = server;
     this.handshaker = handshaker;
     this.metrics = metrics;
-    this.metric = metrics.connected(remoteAddress());
   }
 
   @Override
-  protected Object metric() {
+  protected synchronized Object metric() {
```
Comment (purplefox) [inline src/main/java/io/vertx/core/http/impl/ServerConnection.java]: shouldn't it be metrics(), not metric() ?

Comment (purplefox) [inline src/main/java/io/vertx/core/http/impl/ServerConnection.java]: Typed as Object. really?

**Comment by purplefox** [comment]
Other than the comments, looks fine

File: src/main/java/io/vertx/core/http/impl/ServerConnection.java (hunk 1 above)
Comment (vietj) [inline src/main/java/io/vertx/core/http/impl/ServerConnection.java]: good catch

File: src/main/java/io/vertx/core/http/impl/ServerConnection.java (hunk 2 above)
Comment (vietj) [inline src/main/java/io/vertx/core/http/impl/ServerConnection.java]: yes, this is an handback object provided by the metrics SPI that contains data associated with the connection

**Pull request merged**